src/lib/hesiod.c

```diff
@@ -141,6 +140,12 @@ int hesiod_init(void **context)
 	      else
 		errno = ENOMEM;
 	    }
+	  /* Make sure that the rhs is set. */
+	  if (!ctx->rhs)
+	    {
+	      errno = ENOEXEC;
```

Comment:
Okay, I'm leaning toward EINVAL because at that point we may well have managed to read a configuration file, so ENOENT could be confusing.